packages/context-editor/src/components/SuggestPanel.tsx

```diff
@@ -1,12 +1,36 @@
-import React, { useEffect, useReducer, useState } from "react";
+import React, { useEffect, useState } from "react";
+import { useEditorEffect } from "@handlewithcare/react-prosemirror";
 import { RectangleEllipsis, StickyNote, ToyBrick } from "lucide-react";
 
-import { Card, CardContent } from "ui/card";
-
 import type { SuggestProps } from "../ContextEditor";
+import { reactPropsKey } from "../plugins/reactProps";
 
-export default function SuggestPanel({ isOpen, selectedIndex, items, filter }: SuggestProps) {
+type Props = {
+	suggestData: SuggestProps;
+	setSuggestData: any;
+};
+export default function SuggestPanel({ suggestData, setSuggestData }: Props) {
+	const { isOpen, selectedIndex, items, filter } = suggestData;
 	const [position, setPosition] = useState([0, 0]);
+
+	/**
+	 * In order to get the suggestions to the plugin, we pass props through
+	 * reactPropsKey which the plugin can then access.
+	 */
```

Comment:
this was the only part I couldn't get to be totally independent of passing state around plugins. I don't think `react-prosemirror` offers anything for passing state between plugins and react components, so I'm using the same `reactPropsKey` method we were using before. The `contextSuggestion` plugin uses `prosemirror-autocomplete` so I'm not very confident in tweaking it